src/scheduler.rs

```diff
@@ -57,16 +64,10 @@ async fn scheduler_loop(
     region: String,
     redis_enable_cluster: bool,
     redis_timeouts_ms: u64,
-) {
-    let client = build_redis_client(&redis_host, redis_enable_cluster);
-    let mut connection = client
-        .get_async_connection(redis_timeouts_ms)
-        .await
-        .expect("Unable to connect to redis");
-    let last_progress: Option<String> = connection
-        .get(&progress_key)
-        .await
-        .expect("Unable to get last tick key");
+) -> anyhow::Result<()> {
+    let client = build_redis_client(&redis_host, redis_enable_cluster)?;
+    let mut connection = client.get_async_connection(redis_timeouts_ms).await?;
+    let last_progress: Option<String> = connection.get(&progress_key).await?;
```

Comment:
If these error, is it possible to annotate them with what happened? I think it'd be more helpful to know that redis failed when we attempted to get the key, for example. I guess we'll also know that from the stack trace though